channels/sync_runner.go

```diff
@@ -260,12 +260,16 @@ func AccessNameToPrincipalName(accessPrincipalName string) (principalName string
 func ottoValueToStringArray(value otto.Value) []string {
 	nativeValue, _ := value.Export()
 
-	result := base.ValueToStringArray(nativeValue)
+	result, nonStrings := base.ValueToStringArray(nativeValue)
 
-	if result == nil && !value.IsNull() && !value.IsUndefined() {
-		base.Warnf("SyncRunner: Non-string, non-array passed to JS callback: %s", base.UD(value))
+	if !value.IsNull() && !value.IsUndefined() {
+		if result == nil {
+			base.Warnf("SyncRunner: Non-string, non-array passed to JS callback: %s", base.UD(value))
+		}
+		if nonStrings != nil {
+			base.Warnf("SyncRunner: Non-string slice values: %s passed to JS callback will be omitted", base.UD(nonStrings))
```

Comment:
I think we should rephrase all of the warnings to be more explicit about what it means to the end-user.

Something like, `"Channel names must be string values only. Ignoring non-string channels: %s", nonStrings"`

Reply:
Yeah, makes more sense. This change has been committed.